src/path_rewriting.rs

```diff
@@ -40,14 +40,36 @@ pub fn canonicalize_path<P: AsRef<Path>>(path: P) -> io::Result<PathBuf> {
     Ok(path)
 }
 
-fn normalize_path<P: AsRef<Path>>(path: P) -> PathBuf {
-    let mut new_path = PathBuf::from("");
-    for component in path.as_ref().components() {
-        if component != Component::CurDir {
-            new_path.push(component);
+
+pub fn normalize_path<P: AsRef<Path>>(path: P) -> Option<PathBuf> {
+    // Copied from Cargo sources: https://github.com/rust-lang/cargo/blob/master/src/cargo/util/paths.rs#L65
+    let mut components = path.as_ref().components().peekable();
+    let mut ret = if let Some(c @ Component::Prefix(..)) = components.peek().cloned() {
+        components.next();
+        PathBuf::from(c.as_os_str())
+    } else {
+        PathBuf::new()
+    };
+
+    for component in components {
+        match component {
+            Component::Prefix(..) => unreachable!(),
+            Component::RootDir => {
+                ret.push(component.as_os_str());
+            }
+            Component::CurDir => {}
+            Component::ParentDir => {
+                if !ret.pop() {
+                    eprintln!("Warning: {:?} cannot be normalized because of \"..\", so skip it.", path.as_ref());
+                    return None;
```

Comment:
Maybe we should still normalize it ('../foo/bar' -> 'foo/bar'), but printing the warning so the user knows that something might be wrong.
This way at least we don't totally ignore a file, even if we might be attributing its coverage to a wrong file.

Reply:
If we ignore the path -> we don't risk attributing the coverage to a wrong file.
If we don't ignore the path -> we don't ignore the coverage of the file.

I'm not sure which one of the two is preferable...

Reply:
I think it's really a mistake to change "../foo/bar" into "foo/bar" because the two paths are different.
I think in such a case, the user should use source_dir option to be able to prepend the source_dir path to any too much relative paths... but right now source_dir needs to exist (see main.rs).

Reply:
OK, let's do 1 then.